trio/_core/_asyncgens.py

```diff
@@ -22,17 +29,17 @@ class AsyncGenerators:
     # asyncgens after the system nursery has been closed, it's a
     # regular set so we don't have to deal with GC firing at
     # unexpected times.
-    alive = attr.ib(factory=weakref.WeakSet)
+    alive: weakref.WeakSet[AsyncGenerator[Any, Any]] | set[AsyncGenerator[Any, Any]] = attr.ib(factory=weakref.WeakSet)  # type: ignore[assignment]
```

Comment:
Instead of a union, this could just be `collections.abc.MutableSet`, which both sets inherit from.

Reply:
Changed in 0205c8f

Reply:
Reverted in 2e255eb